### ISSUE TITLE
Filtering issues by label in a specific milestone is not possible

### ISSUE THREAD

**Issue by ClintNase** [webcompat/webcompat.com#3720]
**Url:** https://webcompat.com/issues

**Environment:**
Operating system: Windows 10
Firefox version: Nightly 104.0a1 (2022-07-17) (64-bit)

**Prerequisites:** Must be logged in on Github

**Steps to reproduce:**
1. Go to https://webcompat.com/issues
2. Click on any milestone in order to select it
3. Type any label in the search bar (eg. "label:engine-gecko")
4. Observe the result

**Expected Behavior:**
Filtering issues by label in a specific milestone is possible.

**Actual Behavior:**
Filtering issues by label in a specific milestone is not possible, the label returns issues across all the milestones. The milestone is deselected.

**Notes:**
1. Video provided https://gifyu.com/image/SNpZ6
2. Reproducible on Chrome as well
3. Selecting a different milestone and applying the same label won't do anything (works if you apply a different label)

**Watchers:**
@softvision-oana-arbuzov
@softvision-raul-bucata
@ClintNase

**Comment by karlcow** [comment]
It goes from 
https://webcompat.com/issues?page=1&per_page=50&state=open&stage=needsdiagnosis&sort=created&direction=desc

to 

https://webcompat.com/issues?page=1&per_page=50&state=open&stage=all&sort=created&direction=desc&q=label%3Aengine-gecko

`stage` parameter switched from `needsdiagnosis` to `all`

**Comment by karlcow** [comment]
https://github.com/webcompat/webcompat.com/blob/67f2831bd6a12a157a44ec165590f0b8edb03c31/webcompat/static/js/lib/models/issue.js#L154-L167
